app/services/glimr_new_case.rb

```diff
@@ -0,0 +1,73 @@
+require 'glimr_api_client'
+
+class GlimrNewCase
+  attr_reader :tribunal_case, :case_reference, :confirmation_code
+  alias_attribute :tc, :tribunal_case
+
+  def initialize(tribunal_case)
+    @tribunal_case = tribunal_case
+  end
+
+  def call!
+    response = GlimrApiClient::RegisterNewCase.call(params)
+    @case_reference = response.response_body[:tribunalCaseNumber]
+    @confirmation_code = response.response_body[:confirmationCode]
+  end
+
+  # TODO: find out where to obtain all the required information
+  def params
+    {
+      jurisdictionId: jurisdiction_id,
+      onlineMappingCode: tc.case_type,
+      contactFirstName: taxpayer_first_name,
```

Comment:
The way we have it right now, first/last name are only defined for individuals, not companies. Does GLiMR require these fields for non-individual appellants?